# What is the True Normal Human Body Temperature? 

#### Background

The mean normal body temperature was held to be 37$^{\circ}$C or 98.6$^{\circ}$F for more than 120 years since it was first conceptualized and reported by Carl Wunderlich in a famous 1868 book. But, is this value statistically correct?

<div class="span5 alert alert-info">
<h3>Exercises</h3>

<p>In this exercise, you will analyze a dataset of human body temperatures and employ the concepts of hypothesis testing, confidence intervals, and statistical significance.</p>

<p>Answer the following questions <b>in this notebook below and submit to your Github account</b>.</p> 

<ol>
<li>  Is the distribution of body temperatures normal? 
    <ul>
    <li> Although this is not a requirement for CLT to hold (read CLT carefully), it gives us some peace of mind that the population may also be normally distributed if we assume that this sample is representative of the population.
    </ul>
<li>  Is the sample size large? Are the observations independent?
    <ul>
    <li> Remember that this is a condition for the CLT, and hence the statistical tests we are using, to apply.
    </ul>
<li>  Is the true population mean really 98.6 degrees F?
    <ul>
    <li> Would you use a one-sample or two-sample test? Why?
    <li> In this situation, is it appropriate to use the $t$ or $z$ statistic? 
    <li> Now try using the other test. How is the result be different? Why?
    </ul>
<li>  At what temperature should we consider someone's temperature to be "abnormal"?
    <ul>
    <li> Start by computing the margin of error and confidence interval.
    </ul>
<li>  Is there a significant difference between males and females in normal temperature?
    <ul>
    <li> What test did you use and why?
    <li> Write a story with your conclusion in the context of the original problem.
    </ul>
</ol>

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

#### Resources

+ Information and data sources: http://www.amstat.org/publications/jse/datasets/normtemp.txt, http://www.amstat.org/publications/jse/jse_data_archive.htm
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

****

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as stats
%matplotlib inline 
df = pd.read_csv('data/human_body_temperature.csv')
temp = df.temperature
heart = df.heart_rate

In [2]:
# Your work here.
df.head()

temperature gender  heart_rate
0         99.3      F        68.0
1         98.4      F        81.0
2         97.8      M        73.0
3         99.2      F        66.0
4         98.0      F        73.0

# Q1 Is the distribution of body temperatures normal?
# Although this is not a requirement for CLT to hold (read CLT carefully), it gives us some peace of mind that the population may also be normally distributed if we assume that this sample is representative of the population.

In [3]:
stats.normaltest(temp.tolist())

NormaltestResult(statistic=2.7038014333192031, pvalue=0.2587479863488254)

yes it is 

# Q2 Is the sample size large? Are the observations independent?
# Remember that this is a condition for the CLT, and hence the statistical tests we are using, to apply.

In [24]:
temp.count()

130

130 is not that big of a sample

# Q3 Is the true population mean really 98.6 degrees F?
# Would you use a one-sample or two-sample test? Why?
# In this situation, is it appropriate to use the  t or  z statistic?
 # Now try using the other test. How is the result be different? Why?

In [5]:
mean,std = temp.mean(), temp.std()

95% confidence it is

In [37]:
stats.ttest_1samp(temp, 98.6)

Ttest_1sampResult(statistic=-5.4548232923645195, pvalue=2.4106320415561276e-07)

no it is not 

# Q4 At what temperature should we consider someone's temperature to be "abnormal"?
# Start by computing the margin of error and confidence interval.


In [7]:
np.percentile(temp, [2.5, 97.5])

array([ 96.7225,  99.4775])

anything below 96.72 and above 99.48

# Q5 Is there a significant difference between males and females in normal temperature?
# What test did you use and why?
# Write a story with your conclusion in the context of the original problem.

In [28]:
maletemp = df[df.gender=='M'].temperature
femaletemp = df[df.gender=='F'].temperature

In [39]:
stats.ttest_ind(maletemp, femaletemp)

Ttest_indResult(statistic=-2.2854345381656103, pvalue=0.023931883122395609)

it is more than 1% so they are not that different